src/DataTypes/Serializations/SerializationArray.cpp

```diff
@@ -375,6 +378,19 @@ void SerializationArray::deserializeBinaryBulkWithMultipleStreams(
         else
             SerializationNumber<ColumnArray::Offset>().deserializeBinaryBulk(column_array.getOffsetsColumn(), *stream, 0, rows_offset + limit, 0);
 
+        /// Verify offsets if the data comes over the network
```

Comment:
Few questions:
* Do you know if we do or need other validations, like checking the offsets finish where the data finishes and so on?
* Is this the only path data from network might take to deserialize into an Array?
* Can we add some minor test like sending malformed data and verify it fails? I guess we can do something like `echo "" | xxd | curl`

Reply:
> Do you know if we do or need other validations, like checking the offsets finish where the data finishes and so on?

There are bunch of checks already, but this one was missing. So AFAICS after this one it will not be possible to INSERT incorrect data that will pops up only during reading the column

> Is this the only path data from network might take to deserialize into an Array?

Array - yes.

> Can we add some minor test like sending malformed data and verify it fails? I guess we can do something like echo "" | xxd | curl

Test added and undrafted